signer/xpi/xpi.go

```diff
@@ -157,6 +187,19 @@ func New(conf signer.Configuration, stats *signer.StatsClient) (s *XPISigner, er
 	s.Mode = conf.Mode
 	s.stats = stats
 
+	if conf.Mode == ModeAddOnWithRecommendation {
+		s.recommendationAllowedStates = conf.RecommendationConfig.AllowedStates
+		s.recommendationValidityRelativeStart = conf.RecommendationConfig.ValidityRelativeStart
+		s.recommendationValidityDuration = conf.RecommendationConfig.ValidityDuration
+		log.Infof("xpi: set recommendation options allowed states: %+v, path: %s, start duration: %s duration: %s",
+			s.recommendationAllowedStates,
+			s.recommendationFilePath,
+			s.recommendationValidityRelativeStart,
+			s.recommendationValidityDuration)
+	}
+	s.recommendationFilePath = conf.RecommendationConfig.FilePath
+	log.Infof("xpi: signer %s is ignoring recommendation file path %s", s.ID, s.recommendationFilePath)
```

Comment:
nit: use `%q` for strings so we know when an empty string is logged (it'll set `""`)

Reply:
why is it `ignoring` the recommendation file path ?

Reply:
https://github.com/mozilla-services/autograph/pull/295#discussion_r289018352